### PR TITLE
Add code and tests for thread_notifications_dispatcher

### PR DESCRIPTION
**Pull request by dcristoloveanu**

### CODE REVIEW

File: tests/reals/real_tcall_dispatcher_thread_notification_call.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#include "clds/tcall_dispatcher_ll.h"
+#include "clds/tcall_dispatcher.h"
+#include "thread_notifications_lackey_dll/thread_notifications_lackey_dll.h"
+
+#include "../../inc/clds/tcall_dispatcher_thread_notification_call.h"
+
+#define R2(X) REGISTER_GLOBAL_MOCK_HOOK(X, real_##X);
+
+#define REGISTER_TCALL_DISPATCHER_THREAD_NOTIFICATION_CALL_GLOBAL_MOCK_HOOK() \
+    REGISTER_GLOBAL_MOCK_HOOK(TCALL_DISPATCHER_REGISTER_TARGET(THREAD_NOTIFICATION_CALL), TCALL_DISPATCHER_LL_REGISTER_TARGET(real_THREAD_NOTIFICATION_CALL)) \
```
Comment (anporumb) [inline tests/reals/real_tcall_dispatcher_thread_notification_call.h:18]: >_LL [](http://example.com/codeflow?start=106&length=3)

LL shouldn't be needed here...

File: tests/reals/real_thread_notifications_dispatcher_renames.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// Licensed under the MIT license.See LICENSE file in the project root for full license information.
```
Comment (anporumb) [inline tests/reals/real_thread_notifications_dispatcher_renames.h:1]: >Licensed [](http://example.com/codeflow?start=3&length=8)

missing // Copyright (c) Microsoft. All rights reserved.

here and everywhere

File: tests/thread_notifications_dispatcher_int/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+${theseTestsName}.c
+)
+
+set(${theseTestsName}_c_files
+)
+
+set(${theseTestsName}_h_files
+)
+
+build_test_artifacts(${theseTestsName} "tests/clds" ADDITIONAL_LIBS c_pal c_pal_reals c_util c_util_reals clds_reals clds)
```
Comment (anporumb) [inline tests/thread_notifications_dispatcher_int/CMakeLists.txt:15]: >c_pal_reals [](http://example.com/codeflow?start=74&length=11)

why are "reals" needed in the context of a INT test?

**Review by anporumb — APPROVED** [review]
:shipit:

File: tests/reals/real_tcall_dispatcher_thread_notification_call.h (hunk 1 above)
Comment (dcristoloveanu) [reply tests/reals/real_tcall_dispatcher_thread_notification_call.h:18]: Yeah, build with the HL. Was copied from TARRAY lol.

Comment (dcristoloveanu) [reply tests/reals/real_tcall_dispatcher_thread_notification_call.h:18]: Fixed.

File: tests/reals/real_thread_notifications_dispatcher_renames.h (hunk 2 above)
Comment (dcristoloveanu) [reply tests/reals/real_thread_notifications_dispatcher_renames.h:1]: Fixed.

File: tests/thread_notifications_dispatcher_int/CMakeLists.txt (hunk 3 above)
Comment (dcristoloveanu) [reply tests/thread_notifications_dispatcher_int/CMakeLists.txt:15]: Not needed, copy pasta from when the project was cloned. Removed now.

File: inc/clds/tcall_dispatcher_thread_notification_call.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
```
Comment (jebrando) [inline inc/clds/tcall_dispatcher_thread_notification_call.h:1]: >. [](http://example.com/codeflow?start=59&length=1)

Do you need to put the licence comment here?

**Review by jebrando — APPROVED** [review]
:shipit:

File: inc/clds/tcall_dispatcher_thread_notification_call.h (hunk 4 above)
Comment (dcristoloveanu) [reply inc/clds/tcall_dispatcher_thread_notification_call.h]: How did you see it? Fixed!

**Review by jebrando — APPROVED** [review]
:shipit:

**Pull request merged**